src/services/eventService.js

```diff
@@ -169,6 +169,55 @@ async function getByMonth(startDate, endDate) {
     return events;
 }
 
+async function getUpcomingEvents(today) {
+    let query = {
+        isDeleted: false,
+        isApproved: true
+    };
+
+    if (today?.todayStart) {
+        // Query for upcoming events
+        // An event is upcoming if any of its dates are on or after todayEnd
+        query.dates = {
+            $elemMatch: {
+                date: { $gte: today.todayStart }
+            }
+        };
+
+    } else {
+        // Handle invalid 'today' parameter
+        return []; // Or handle as needed
+    }
+
+    const events = await Event.find(query);
+    return events;
+}
+async function getPastEvents(today) {
+    let query = {
+        isDeleted: false,
+        isApproved: true
+    };
+
+    if (today?.todayStart) {
+        // Query for past events
+        // An event is past if all of its dates are before todayStart
+        query.dates = {
+            $not: {
+                $elemMatch: {
+                    date: { $gte: today.todayStart }
+                }
+            }
+        };
+
+    } else {
+        // Handle invalid 'today' parameter
```

Comment:
Not handled?

Reply:
This comment is for clarification, ok i will remove it.